config/application.rb

```diff
@@ -31,6 +32,16 @@ class Application < Rails::Application
 
     # config.middleware.use Rack::Deflater
 
+    # Initialize GraphQL client for Github API v4
+    # Load schema from previous Github API schema dump
+    Schema = Application.root.join("config/github_graphql_schema.json").to_s
```

Comment:
What happens when it changes?

Reply:
I was initially thinking of having a rake task to update this, but now that I think through it more that might not work out. I can try and mess with the file to see what it would do if it is not up to date. We may want to store this in a shared location and have a script pull it down and reupload it to the shared area when needed. The alternative is to update this on startup of the application which would require a valid Github token available during initialization.

Reply:
crashing app startup if we can't load this schema over the network strikes me as :-1: 
also we'd kinda like to notice if it changes by seeing that change in a git diff, I think? Sort of like a manifest vs. a lockfile (manifest = the canonical schema on github's site, lockfile = our local copy)

Reply:
After some testing I think the schema is mostly used to parse and verify any GraphQL queries declared in the app. So I think there are 2 cases to think about when the local schema differs from the Github API:

1. GH updates their schema and adds fields, types, connections, etc.
In this case, we may potentially be missing out on a new feature but our current queries would be fine and still be accepted by the API.

2. GH updates their schema and removes something we are using in a query.
In this case, this means our queries will still be considered valid by the GraphQL client and the app would still start up correctly. However we would receive errors back in the response from the API. We should handle the error case anyways to help notify on query problems. We would need to update the `schema.json` file and update our queries.

Reply:
Okay, given this it seems like storing is the best of the bad options. *sigh* I thought graphql was supposed to be better than this!

We should have a rake task we can use to update the file though